iceoryx_utils/source/posix_wrapper/timer.cpp

```diff
@@ -76,52 +76,93 @@ void Timer::OsTimer::callbackHelper(sigval data)
         return;
     }
 
-    handle.m_timerInvocationCounter.fetch_add(1u, std::memory_order_relaxed);
+    // signal that we intend to call the callback (but we may not start it ourself, it may be done
+    // by another invocation of this function)
+    auto invocationCounter = handle.m_timerInvocationCounter.fetch_add(1u, std::memory_order_relaxed);
 
-    do
+    // avoid the spurious failures of try_lock with this flag end still reduce contention
+    // if another thread is in this section we return but we already stated our intention to call the callback
+    // via m_timerInvocationCounter and the other thread will do this for us
+    // note that thte invocation may be lost if the callback is already running, this is intentional
+    // to avoid an unlimited number of callback invocations to pile up
+
+    // note: if we can tolerate those calls to pile up we can get rid of this flag and the while loop as well
+
+    if (!handle.m_callbackIsAboutToBeExecuted.test_and_set(std::memory_order_acq_rel))
     {
-        if (handle.m_accessMutex.try_lock() == false)
+        // flag protected region
+
+        // the mutex is needed to protect against concurrent deletion of the timer in the destructor
+        std::lock_guard<std::mutex> lockGuard(handle.m_accessMutex);
+
+        // clear the flag regardless of how we leave the function
+        cxx::GenericRAII clearCallbackFlag(
+            []() {}, [&]() { handle.m_callbackIsAboutToBeExecuted.clear(std::memory_order_release); });
+
+        do
         {
-            if (handle.m_catchUpPolicy == CatchUpPolicy::TERMINATE)
+            // prohibits other threads from entering the flag protected region
+            handle.m_callbackIsAboutToBeExecuted.test_and_set(std::memory_order_acq_rel);
+
+            if (handle.m_timer == nullptr)
             {
-                errorHandler(Error::kPOSIX_TIMER__CALLBACK_RUNTIME_EXCEEDS_RETRIGGER_TIME);
+                errorHandler(Error::kPOSIX_TIMER__INCONSISTENT_STATE);
+                return;
```

Comment:
The complaints of the Codecov are valid (i.e. it is true), but it is also hard to inject the faults leading into those branches. We may need a general fault injection strategy for scenarios like this. Part of the error handling strategy I guess.